generator/src/test/java/com/scottlogic/deg/generator/cucumber/longerThan.feature

```diff
@@ -0,0 +1,671 @@
+      | 2019-01-01T00:00:00.001   |
+
+
+  Scenario: Running an 'longerThan' request alongside a  afterOrAt test2  constraint should be successful
+    Given foo is longer than 1
+    And foo is anything but after or at 2019-01-01T00:00:00.000
+    And foo is in set:
+      | "a"    |
+      | "aa"   |
+      | "aaaa" |
+      | "aaaaaaaaaa" |
+      | "2018-02-01T00:00:00.001" |
+      | "2019-01-01T00:00:00.000" |
+      | 2018-01-01T00:00:00.001   |
+      | 2019-01-01T00:00:00.000   |
+    Then the following data should be generated:
+      | foo    |
+      | null   |
+      | "aa"   |
+      | "aaaa" |
+      | "aaaaaaaaaa" |
+      | "2018-02-01T00:00:00.001" |
+      | "2019-01-01T00:00:00.000" |
+      |  2018-01-01T00:00:00.001  |
+
+  Scenario: Running a 'longerThan' request using a number (decimal number) to specify a the length of a generated string should fail with an error message
+    Given foo is longer than 1.1
+    Then I am presented with an error message
+    And no data is created
+
+  Scenario: Running a 'longerThan' request alongside a before constraint should be successful
+    Given foo is longer than 1
+    And foo is before 2019-01-01T00:00:00.000
+    And foo is in set:
+      | "a"  |
+      | "aa"  |
+      | "aaaaa" |
+      | "2019-01-01T00:00:00.000" |
+      | 2018-01-01T00:00:00.000 |
+      | 2019-01-01T00:00:00.000 |
+    Then the following data should be generated:
+      | foo     |
+      | null    |
+      | "aa"  |
+      | "aaaaa" |
+      | "2019-01-01T00:00:00.000" |
+      | 2018-01-01T00:00:00.000 |
+
+  Scenario: Running a 'longerThan' request alongside a before test 2 constraint should be successful
+    Given foo is longer than 1
+    And foo is anything but before 2019-01-01T00:00:00.000
+    And foo is in set:
+      | "a"  |
+      | "aa"  |
+      | "aaaaa" |
+      | "2019-01-01T00:00:00.000" |
+      | 2018-01-01T00:00:00.000 |
+      | 2019-02-01T00:00:00.000 |
+      | 2020-02-01T00:00:00.000 |
+    Then the following data should be generated:
+      | foo     |
+      | null    |
+      | "aa"  |
+      | "aaaaa" |
+      | "2019-01-01T00:00:00.000" |
+      | 2019-02-01T00:00:00.000 |
+      | 2020-02-01T00:00:00.000 |
+
+Scenario: Running a 'longerThan' request alongside a non contradicting  beforeOrAt constraint should be successful
+  Given foo is longer than 1
+  And foo is before or at 2019-01-01T00:00:00.000
+  And foo is in set:
+    | "a"  |
+    | "aa"  |
+    | "aaaa" |
+    | "2019-01-01T00:00:00.000" |
+    | 2019-01-01T00:00:00.000   |
+    | 2018-01-01T00:00:00.000   |
+    | 2020-01-01T00:00:00.000   |
+  Then the following data should be generated:
+    | foo     |
+    | null    |
+    | "aa"    |
+    | "aaaa"  |
+    | "2019-01-01T00:00:00.000" |
+    | 2019-01-01T00:00:00.000   |
+    | 2018-01-01T00:00:00.000   |
+
+Scenario: Running a 'longerThan' request alongside a non contradicting  beforeOrAt test 2 constraint should be successful
```

Comment:
remove 'test 2' from scenario name and rename uniquely.

Reply:
Yep, renamed it .